.github/workflows/framework-tests.yaml

```diff
@@ -38,6 +38,7 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest]
         python-version: ["3.8", "3.10"]
+        root: [true, false]
```

Comment:
I don't love how this shows up in the test summary in GitHub, e.g., "ubuntu-latest, 3.10, false" -- it's unclear what "false" means. Also, I'd rather not run all the tests twice, and I don't think we need to or should be running all the tests as root anyway.

Can we use a suitable test name prefix, and then do something like we do with the RealPebble tests below? Perhaps a TestFilesystemAsRoot class and then use `-k FilesystemAsRoot` filter. I think having a separate job is cleaner (even though it's a bit more duplication).

Reply:
I think the tests depend on the outcome of the discussion, I will defer the changes for now.